src/misc/setup.cpp

```diff
@@ -807,91 +943,130 @@ string Section_prop::GetPropValue(const std::string &_property) const
 {
 	return NO_SUCH_PROPERTY;
 }
 
-bool Config::PrintConfig(const std::string &filename) const
+bool Config::PrintConfig(const std::string& filename) const
 {
 	char temp[50];
 	char helpline[256];
-	FILE *outfile = fopen(filename.c_str(), "w+t");
-	if (outfile == NULL) return false;
 
-	/* Print start of configfile and add a return to improve readibility. */
+	FILE* outfile = fopen(filename.c_str(), "w+t");
+	if (outfile == NULL) {
+		return false;
+	}
+
+	// Print start of config file and add a return to improve readibility
 	fprintf(outfile, MSG_GetRaw("CONFIGFILE_INTRO"), VERSION);
 	fprintf(outfile, "\n");
 
 	for (auto tel = sectionlist.cbegin(); tel != sectionlist.cend(); ++tel) {
-		/* Print out the Section header */
+		// Print section header
 		safe_strcpy(temp, (*tel)->GetName());
 		lowcase(temp);
-		fprintf(outfile,"[%s]\n",temp);
+		fprintf(outfile, "[%s]\n", temp);
 
-		Section_prop *sec = dynamic_cast<Section_prop *>(*tel);
+		Section_prop* sec = dynamic_cast<Section_prop*>(*tel);
 		if (sec) {
-			Property *p;
-			int i = 0;
+			Property* p;
+			int i           = 0;
 			size_t maxwidth = 0;
+
 			while ((p = sec->Get_prop(i++))) {
 				maxwidth = std::max(maxwidth, p->propname.length());
 			}
-			i=0;
+
+			i = 0;
+
 			char prefix[80];
 			int intmaxwidth = std::min<int>(60, check_cast<int>(maxwidth));
 			safe_sprintf(prefix, "\n# %*s  ", intmaxwidth, "");
-			while ((p = sec->Get_prop(i++))) {
 
-				if (p->IsDeprecated())
+			while ((p = sec->Get_prop(i++))) {
+				if (p->IsDeprecated()) {
 					continue;
+				}
 
 				std::string help = p->GetHelpUtf8();
+
 				std::string::size_type pos = std::string::npos;
-				while ((pos = help.find('\n', pos+1)) != std::string::npos) {
+
+				while ((pos = help.find('\n', pos + 1)) !=
+				       std::string::npos) {
 					help.replace(pos, 1, prefix);
 				}
 
-				fprintf(outfile, "# %*s: %s", intmaxwidth, p->propname.c_str(), help.c_str());
+				fprintf(outfile,
+				        "# %*s: %s",
+				        intmaxwidth,
+				        p->propname.c_str(),
+				        help.c_str());
 
 				std::vector<Value> values = p->GetValues();
+
 				if (!values.empty()) {
-					fprintf(outfile, "%s%s:", prefix, MSG_GetRaw("CONFIG_SUGGESTED_VALUES"));
-					std::vector<Value>::const_iterator it = values.begin();
+					fprintf(outfile,
+					        "%s%s:",
+					        prefix,
+					        MSG_GetRaw("CONFIG_VALID_VALUES"));
+
+					std::vector<Value>::const_iterator it =
+					        values.begin();
+
 					while (it != values.end()) {
-						if((*it).ToString() != "%u") { //Hack hack hack. else we need to modify GetValues, but that one is const...
-							if (it != values.begin()) fputs(",", outfile);
-							fprintf(outfile, " %s", (*it).ToString().c_str());
+						if ((*it).ToString() !=
+						    "%u") { // Hack hack hack.
+							    // else we need to
+							    // modify GetValues,
+							    // but that one is
+							    // const...
+							if (it != values.begin()) {
```

Comment:
Just commenting on white-space use here (no the code), but these unwrapped comments at the far-right edge of the code are painful..

![2023-03-22_11-15](https://user-images.githubusercontent.com/1557255/227000165-58fe4fb9-a485-4fd3-8435-3d843a7bd637.png)
 
I much prefer the commend just be on the line directly above the code, and maybe separate with a new line above it so the commend and code are together.

``` c++

			// Hack hack hack. else we need to modify GetValues,
			// but that one is const...
			if((*it).ToString() != "%u") { 
```

So I really appreciate clang-format wrangling these back "inside the box"!

Reply:
Yeah good points, I've cleaned up the comments somewhat.